src/worker.js

```diff
@@ -285,9 +285,9 @@ self.onmessage = (e) => {
       }
     } else if (e.data.target === 'setimmediate') {
       // no-op
-    } else if (e.data.cmd === 'processThreadQueue') {
+    } else if (e.data.cmd === 'processProxyingQueue') {
       if (Module['_pthread_self']()) { // If this thread is actually running?
-        Module['_emscripten_current_thread_process_queued_calls']();
+        Module['_emscripten_proxy_execute_queue'](e.data.queue);
```

Comment:
Do we need to pass the queue pointer around or can we just assume its the system queue here?

Reply:
Yes, we need to pass the queue pointer around because this is the same code path used for both the system queue and user queues. Actually, I see that this ended up duplicated somehow as well. Will fix.